### PR TITLE
update submodule libuv to v.1.34.1

### PR DESCRIPTION
**Pull request by zhaozg**

### CODE REVIEW

File: tests/test-fs.lua [hunk 1]
```diff
@@ -246,4 +246,34 @@ return require('lib/tap')(function (test)
...
+    end)
+  end)
+
+  test("fs.mkdtemp sync error", function(print, p, expect, uv)
+    local tp = "luvXXXXXZ"
+    local path, err, code = uv.fs_mkdtemp(tp)
+    assert(path==nil)
+    assert(err:match("^EEXIST:"))
+    assert(code=='EEXIST')
+  end)
```
Comment (squeek502) [inline tests/test-fs.lua]: I'm getting EINVAL errors for these on Windows.

Comment (squeek502) [inline tests/test-fs.lua]: I'm getting `EINVAL` errors for these on Windows.

File: tests/test-fs.lua [hunk 2]
```diff
@@ -246,4 +246,34 @@ return require('lib/tap')(function (test)
...
+      assert(not err)
+      assert(path:match("^luv......"))
+    end)
+  end)
+
+  test("fs.mkdtemp sync", function(print, p, expect, uv)
+    local tp = "luvXXXXXX"
+    local path, err, code = uv.fs_mkdtemp(tp)
+    assert(path:match("^luv......"))
+  end)
```
Comment (squeek502) [inline tests/test-fs.lua]: These actually create the directories so they need to be cleaned up afterwards.

**Comment by squeek502** [comment]
Made a PR for `uv_random` bindings against this branch here: https://github.com/zhaozg/luv/pull/1

File: src/tty.c [hunk 3]
```diff
@@ -60,3 +60,38 @@ static int luv_tty_get_winsize(lua_State* L) {
...
+    case UV_TTY_SUPPORTED:
+      lua_pushliteral(L, "supported");
+      ret = 1;
+      break;
+    case UV_TTY_UNSUPPORTED:
+      lua_pushliteral(L, "unsupported");
+      ret = 1;
+      break;
+    default:
+      lua_pushinteger(L, state);
```
Comment (squeek502) [inline src/tty.c]: It might be better to throw an error here? Maybe something like: `luaL_error(L, "unexpected uv_tty_vtermstate: %d", state);`

**Comment by zhaozg** [comment]
> Nice work, should be good to go if all the tests pass. 👍

Thank you very much.

**Comment by squeek502** [comment]
`fs.mkdtemp async error` test is failing on Mac: https://travis-ci.org/luvit/luv/jobs/638359999#L517

`Uncaught Error: ./tests/test-fs.lua:269: attempt to index a nil value (local 'err')`

**Pull request merged**